libdnf/goal/Goal.cpp

```diff
@@ -1436,7 +1441,7 @@ Goal::Impl::protectedInRemovals()
         if (id == -1)
             break;
 
-        if (protectedPkgs->has(id)) {
+        if (protectedPkgs->has(id) || isProtectedRunningKernel(id)) {
```

Comment:
With ProtectedRunningKernel() it could be like
```
    Id protected_kernel = ProtectedRunningKernel()
    while(true) {
        id = removalOfProtected->next(id);
        if (id == -1)
            break;

        if (protectedPkgs->has(id) || id == protected_kernel) {
```